MantidPlot/src/Mantid/MantidUI.cpp

```diff
@@ -3090,11 +3108,28 @@ MultiLayer *MantidUI::plot1D(const QMultiMap<QString, int> &toPlot,
   MantidMatrixCurve::IndexDir indexType =
       (spectrumPlot) ? MantidMatrixCurve::Spectrum : MantidMatrixCurve::Bin;
   MantidMatrixCurve *firstCurve(NULL);
+  QString logValue("");
+  int i = 0;
   for (QMultiMap<QString, int>::const_iterator it = toPlot.begin();
        it != toPlot.end(); ++it) {
+
     try {
-      auto *wsCurve = new MantidMatrixCurve(it.key(), g, it.value(), indexType,
-                                            errs, plotAsDistribution, style);
+      if (!log.isEmpty()) { // Get log value from workspace
+        double logVal;
+        if (customLogValues.size() > 0) {
+          logVal = getSingleWorkspaceLogValue(i++, customLogValues);
```

Comment:
Hmm, I read it wrong. I read `i` as `it`. It is only a counter, that is fine.  This makes more sense now to me.